node/node_test.go

```diff
@@ -18,41 +18,54 @@ import (
 
 var MockServerAddr = ":7980"
 
+// startNode starts the full node and stops it when the test is done
+func startAndStopNode(t *testing.T, node Node) {
+	require.NoError(t, node.Start())
+
+	t.Cleanup(func() {
+		cleanUpNode(node, t)
+	})
+}
+
 // cleanUpNode stops the node and checks if it is running
 func cleanUpNode(node Node, t *testing.T) {
 	assert.NoError(t, node.Stop())
 	assert.False(t, node.IsRunning())
 }
 
-func initializeAndStartFullNode(ctx context.Context, t *testing.T) *FullNode {
-	node := initializeAndStartNode(ctx, t, "full")
-	return node.(*FullNode)
-}
-
-func initializeAndStartLightNode(ctx context.Context, t *testing.T) *LightNode {
-	node := initializeAndStartNode(ctx, t, "light")
-	return node.(*LightNode)
-}
-
-// initializeAndStartNode initializes and starts a test node
-func initializeAndStartNode(ctx context.Context, t *testing.T, nodeType string) Node {
+// initializeAndStartNode initializes and starts a node of the specified type.
+func initializeAndStartNode(ctx context.Context, t *testing.T, nodeType NodeType) Node {
 	node, _ := setupTestNode(ctx, t, nodeType)
+	t.Cleanup(func() {
+		cleanUpNode(node, t)
+	})
```

Comment:
this should be after we start the node, as stopping a node that is already stopped (or isn't started) fails.